test/controllers/v1/profiles_controller_test.rb

```diff
@@ -74,11 +74,20 @@ class DestroyTest < ProfilesControllerTest
       end
 
       test 'destroy an existing, not accessible profile' do
+        profiles(:two).update! parent_profile: profiles(:one)
```

Comment:
This now creates two different cases: (1) delete a not accessible profile, one that exists in the DB, but the user is not authorized by pundit to view the policy at all; (2) delete an accessible profile (accessible via show/index) that the user is not authorized to delete (i.e. a canonical profile).

Case 1 returns a 404 as if the profile does not exist for that user. Case 2 returns a 403 forbidden since the user knows the profile exists but is forbidden to delete it.